src/org/zaproxy/zap/extension/codedx/UploadActionListener.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2016 Code Dx, Inc. - http://www.codedx.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.codedx;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Locale;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.mime.HttpMultipartMode;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.FileBody;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.view.View;
+
+public class UploadActionListener implements ActionListener{
+
+	private static final Logger logger = Logger.getLogger(UploadActionListener.class);
+	
+	private CodeDxExtension extension;
+	private UploadPropertiesDialog prop;
+	
+	public UploadActionListener(CodeDxExtension extension) {
+		this.extension = extension;
+		this.prop = new UploadPropertiesDialog(extension);
+	}
+	
+	@Override
+	public void actionPerformed(ActionEvent e) {
+		prop.openProperties(this);
+	}
+	
+	public void generateAndUploadReport(){
+		String error = null;
+		try {
+			
+			final StringBuilder report = new StringBuilder();
+			final File reportFile = generateReport(report);
+			
+			if(!"".equals(report.toString().trim()) && report.toString().split("\n").length > 2){
```

Comment:
See http://stackoverflow.com/a/3321548/574815 regarding the empty check. Also, you should save `report.toString()` in a variable to avoid calling it twice (as it is a probably relatively expensive operation)